lib/rubygems/commands/uninstall_command.rb

```diff
@@ -148,21 +148,30 @@ def uninstall_all
 
   def uninstall_specific
     deplist = Gem::DependencyList.new
+    original_gem_version = {}
+    gems_to_uninstall = {}
 
     get_all_gem_names_and_versions.each do |name, version|
-      requirement = Array(version || options[:version])
-      gem_specs = Gem::Specification.find_all_by_name(name, *requirement)
+      original_gem_version[name] = requirement = version || options[:version]
+
+      gem_specs = Gem::Specification.find_all_by_name(name, original_gem_version[name])
+
       say("Gem '#{name}' is not installed") if gem_specs.empty?
-      gem_specs.each do |spec|
-        deplist.add spec
-      end
+      gem_specs.each { |spec| deplist.add spec }
```

Comment:
I would leave the style untouched to make the diff of this bug fix simpler.

Reply:
Fair 👍